metadata-ingestion/src/datahub/ingestion/source/bigquery_v2/usage.py

```diff
@@ -218,7 +218,7 @@ def usage_statistics_query(top_n: int) -> str:
                 COUNT(q.query) query_count
             FROM
                 read_events r
-                LEFT JOIN query_events q ON r.name = q.key
+                INNER JOIN query_events q ON r.name = q.key
```

Comment:
Could we end up with incorrect counts because we're basically filtering on a query event existing here, but not when doing user / column counts? e.g. a column or user count > the total query count.

Based on the read reason doc, we probably don't need to count things like materialized view refresh. Perhaps we should just do a delete of all read events not associated with query events?

Reply:
> Could we end up with incorrect counts because we're basically filtering on a query event existing here, but not when doing user / column counts? e.g. a column or user count > the total query count.

Yes. that's a valid concern. "DatasetUsageStatistics" captures number of times table is "used". If we consider query(or SQL query) is just one of the means of usage, keeping column , user counts, as they would be after this PR - makes sense to me.

If this creates confusion, I believe, another way is to fix forward and include the "QueryEvent" for other table read reasons (QUERY_REQUEST, MATERIALIZED_VIEW_REFRESH, CREATE_READ_SESSION, etc). Here "Query" takes a generous meaning than SQL query however.